File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.py
import heapq
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        
        pq = []

        for i in nums:
            heapq.heappush(pq, i)
        
        largest = heapq.nlargest(2, pq)
        
        return (largest[0] - 1) * (largest[1] - 1)

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        
        sort(nums.begin(), nums.end());
        int n = nums.size();

        int last = nums[n - 1];
        int sLast = nums[n - 2];

        return (last - 1) * (sLast - 1);
    }
};